### PR TITLE
chore: add bmt formal verification

### PR DESCRIPTION
**Pull request by ananas-block**
formal verification branch should not be merged at this time.

## Summary by CodeRabbit

* **New Features**
  * Added comprehensive documentation for batched Merkle tree operations including initialization, queue insertions, and tree updates.
  * Introduced formal verification testing infrastructure for enhanced system reliability.
  * Added new error handling for root history management edge cases.

* **Tests**
  * Expanded end-to-end test coverage for address and state tree operations with realistic transaction scenarios.

### CODE REVIEW

File: program-libs/batched-merkle-tree/src/merkle_tree.rs [hunk 1]
```diff
@@ -202,14 +281,35 @@ impl<'a> BatchedMerkleTreeAccount<'a> {
...
+            let ghost_root_batch = GhostRoots {
+                batch_0: Vec::new(),
+                batch_1: Vec::new(),
+            };
+            Ok(BatchedMerkleTreeAccount {
+                pubkey: *pubkey,
+                metadata,
+                root_history,
+                bloom_filter_stores,
+                hash_chain_stores: [hash_chain_store_0, hash_chain_store_1],
```
Comment (ananas-block) [inline program-libs/batched-merkle-tree/src/merkle_tree.rs:309]: use vectors
